_posts/2022-06-22-domain-management-for-multitenant-app-in-aws.markdown

```diff
@@ -0,0 +1,64 @@
+
+### Solution
+
+It depends whether you are hosting your application on cloud or on premise. Cloud hosting has greater flexibility and scalability than on-premise hosting. Hence in this article, I will explain how you can implement your domain management system in AWS. 
+
+The diagram below is the solution to how you can typically start building it, but it will soon become challenging when the number of clients grows.
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureSingleAccount.png)
+
+Generally, it is a good practice to segregate the production environment from the development environment, which should also apply to your domain management system. This approach has no segregation between environments and hosts all domain names within a single location. It increases the risk of impacting all tenants when there are service disruptions. Hence it has a greater chance of affecting the availability of the application. 
+
+The architecture diagram below shows how to manage domains for all tenants in multiple environments to minimize the likelihood of impacting your production environment.  
+
+
+![Domain Management Architecture ](/img/blog/domain-management/architectureDiagram.png)
+
+The critical difference between these two diagrams is the latter has a clear separation between environments. The infrastructure for the non-production environment should be identical to the production environment to enable errors to be caught in the non-production environment first without impacting the production domains. DNS is a public-facing element you would want to minimize downtime as it will directly affect your revenue or reputation.
+
+AWS offers a highly scalable Domain Name System (DNS) Web service called AWS Route53. It provides the ability to segregate the management of domain names between accounts. Having segregation between accounts, especially for non-production and production, is key to reducing the risk of damage to production domains. It is also beneficial for segregating billing costs; hence, you will get better insights. 
+
+Amazon CloudFront is a web service that speeds up the distribution of your static and dynamic web content. You can configure a single CloudFront distribution for each environment in AWS to serve requests from multiple origins or subdomains. It is essential to have SSL certificates configured for the subdomains to secure your web application. An SSL certificate is a digital certificate that authenticates a website's identity and enables an encrypted connection. You can have individual CloudFront distribution and SSL certificates for each subdomain, but having a single distribution and SSL certificate can reduce the overhead for maintenance, such as renewing only one SSL certificate for all subdomains.  
```

Comment:
```suggestion
[Amazon CloudFront](https://aws.amazon.com/cloudfront/), a content delivery network, speeds up the distribution of your static and dynamic web content. You can configure a single CloudFront distribution for each environment in AWS to serve requests from multiple origins or subdomains. It is essential to have [SSL certificates](https://en.wikipedia.org/wiki/Public_key_certificate) configured for each subdomains of your web application.  You can have separate CloudFront distributions, and SSL certificates, for each subdomain, but using a single distribution, and SSL certificate, can reduce the overhead for maintenance. For example, use a single SSL certificate across all subdomains.  
```

Proofreading suggestion.
If the reader has made it this far, you can assume they understand what SSL certificate is - but link out in case.